src/GitHub.VisualStudio/UI/Views/PullRequestDetailView.xaml

```diff
@@ -400,11 +400,21 @@
                     
                     <te:SectionControl.Content>
                         <Grid>
+                            <Grid.Resources>
+                                <!-- We have to do contortions to get to the PullRequestDetailViewModel which contains the OpenFile and DiffFile
+                                     commands: the control which hosts the context menu must have the PullRequestDetailViewModel as its tag,
+                                     which is then set as the tag on the ContextMenu which is finally reacable from the MenuItem. -->
+                                <ContextMenu x:Key="FileContextMenu" Tag="{Binding PlacementTarget.Tag, RelativeSource={RelativeSource Mode=Self}}">
```

Comment:
@grokys  - What you could do is duplicate the commands onto the changed file (so you call the command against the file the the view ViewModel) but then when you set the `ContextMenu`'s `DataContext` to the file also set the command parameter for each `MenuItem` to the views `DataContext` (all in code).

You can then have the command on the file call into the command on the passed in parameter passing itself in, completing the circle.

I'll try and have a play with this tonight once i'm out of work and see if I can get something working nicely, once I can actually play with the code and not just look at a PR.